### ISSUE TITLE
Unable to use the `useAnnotationStore` hook with the `TextAnnotationStore` generic type 

### ISSUE THREAD

**Issue by oleksandr-danylchenko** [recogito/text-annotator-js#18]
When I tried using the `useAnnotationStore` hook, I realized it's bound to the A9S types, including the non-text core `Annotation` type:
```ts
export declare const useAnnotationStore: <T extends Store<Annotation>>() => T;
```
So using methods like `addAnnotation`, `updateTarget`, etc. isn't that convenient. And, unfortunately, when I tried passing the `TextAnnotationStore` type as a generic, I faced the next error: 
![non-compatible types](https://github.com/recogito/text-annotator-js/assets/68850090/7affceb3-d293-4e83-8af7-df99a40276d2)

---

Additionally, I noticed that for the `getBody`, `addBody`, `bulkUpdateBodies`, `updateTarget`, and `bulkUpdateTargets` - the argument type is not generic:
![image](https://github.com/recogito/text-annotator-js/assets/68850090/14444581-3d4b-49df-b791-708b57b48b48)

Which doesn't allow convenient usage of the text-specific bodies/targets. Especially if a specific consuming app has any types of overrides like this:
<img width="429" alt="image" src="https://github.com/recogito/text-annotator-js/assets/68850090/14e8cf2d-19a0-4366-955c-a07cf9317c07">

**Comment by oleksandr-danylchenko** [comment]
Seems like this issue has been seen before. Because in React's `TextAnnotatorPopup`, the store's state is manually coerced: 
https://github.com/recogito/text-annotator-js/blob/054ca425b2957c9184565595ff40bfbd5c26a835/packages/text-annotator-react/src/TextAnnotatorPopup.tsx#L50

**Comment by rsimon** [comment]
Yes, I'm aware. Reasoning is: as soon as you work with the internal state, you are bound to the internal data model. All the API methods that sit on the `anno` object pass through the adapter. But everything that's internal state is going to be internal model. Otherwise I fear complexity might start to get out of control...

**Comment by oleksandr-danylchenko** [comment]
Ahh.. I see... The `Annotator` exposes only methods with the adapter support: 
<img width="693" alt="image" src="https://github.com/recogito/text-annotator-js/assets/68850090/7b432092-8850-4105-854e-ac8b27d3c265">
And the `updateTarget` isn't listed among them...

**Comment by oleksandr-danylchenko** [comment]
Now I'm not even sure whether it was worth spending time on the W3C adapter as the internal structure looks really neat and simple 😅

**Comment by oleksandr-danylchenko** [comment]
But what do you think about at least adding the text-specific `state` to the `TextAnnotator` type?
![image](https://github.com/recogito/text-annotator-js/assets/68850090/89923301-091a-45f6-8e35-756c9a994704)

**Comment by rsimon** [comment]
Ah, right. Sorry I should have read your issue more thoroughly. Yes, of course. The store deals with the inner data model. But, of course, it should work with `TextAnnotation`s, not `Annotation`s. Not sure what the issue is at the moment. But definitely needs fixing.

**Comment by oleksandr-danylchenko** [comment]
Closed in favor of the #47
